contentcuration/contentcuration/static/js/edit_channel/uploader/views.js

```diff
@@ -0,0 +1,484 @@
+var Backbone = require("backbone");
+var _ = require("underscore");
+require("uploader.less");
+var BaseViews = require("./../views");
+var Models = require("./../models");
+
+var AddContentView = BaseViews.BaseListView.extend({
+	template: require("./hbtemplates/add_content_dialog.handlebars"),
+	item_view:"uploading_content",
+	initialize: function(options) {
+		_.bindAll(this, 'add_topic','close_uploader', 'edit_metadata','upload_file','add_file');	
+		this.collection = options.collection;
+		this.main_collection = options.main_collection;
+		this.root = options.root;
+		this.parent_view = options.parent_view;
+		this.render();
+		this.counter = 0;
+	},
+	render: function() {
+		this.$el.html(this.template({
+			folder: this.root,
+			node_list: this.collection.toJSON()
+		}));
+		this.load_content();
+		this.parent_view.set_editing(true);
+	},
+	events: {
+		'click #create_topic':'add_topic',
+		'click .close_uploader' : 'close_uploader',
+		'click .edit_metadata' : 'edit_metadata',
+		'click #upload_file' : 'upload_file',
+		'change input:file' : 'add_file'
+	},
+
+	load_content:function(){
+		var self = this;
+		/* TODO: Use if re-rendering previously selected items*/
+		this.collection.forEach(function(entry){
+			var node_view = new NodeListItem({
+				edit: false,
+				containing_list_view: self,
+				el: self.$el.find("#content_item_" + entry.cid),
+				model: entry,
+				root: this.root
+			});
+			self.views.push(node_view);
+		});
+	},
+	add_topic:function(){
+		$("#upload_content_add_list").append("<div id='new'></div>");
+		var topic = new Models.NodeModel({
+			"kind":"topic", 
+			"title": (this.counter > 0)? "Topic " + this.counter : "Topic",
+			"parent" : this.root.id
+		});
+		console.log("Adding", topic);
+		this.collection.add(topic);
+		this.counter++;
+		var item_view = new NodeListItem({
+			edit: true,
+			containing_list_view: this,
+			el: this.$el.find("#new"),
+			model: topic,
+			root: this.root,
+		});
+		this.views.push(item_view);
+	},
+	close_uploader: function(){
+		this.parent_view.set_editing(false);
+		this.delete_view();
+	},
+	edit_metadata: function(){
+		var metadata_view = new EditMetadataView({
+			collection: this.collection,
+			parent_view: this.parent_view,
+			el: this.$el,
+			root: this.root,
+			allow_add: true,
+			main_collection : this.main_collection
+		});
+		this.undelegateEvents();
+		this.unbind();	
+	},
+	upload_file:function(){
+		this.$el.find("#file-dialog").trigger('click');
+	},
+	add_file:function(){
+		/* TODO: Need to implement uploading files
+		$("#upload_content_add_list").append("<div id='new'></div>");
+		var item_view = new NodeListItem({
+			edit: false,
+			containing_list_view: this,
+			el: this.$el.find("#new"),
+			model: file,
+			root: this.root,
+		});
+		this.views.push(item_view);
+		*/
+	}
+});
+
+var EditMetadataView = BaseViews.BaseListView.extend({
```

Comment:
This would make sense to move out of UploaderViews, as it is not just to do with uploading, and you use it many different places - which I was very glad to see!